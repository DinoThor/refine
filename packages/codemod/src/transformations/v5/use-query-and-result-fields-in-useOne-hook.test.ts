import useQueryAndResultFieldsInUseOneTransform from "./use-query-and-result-fields-in-useOne-hook";
import jscodeshift, { type JSCodeshift } from "jscodeshift";

const transform = (source: string) => {
  const j: JSCodeshift = jscodeshift.withParser("tsx");
  const root = j(source);

  return useQueryAndResultFieldsInUseOneTransform(j, root);
};

describe("use-query-and-result-fields-in-useOne-hook", () => {
  it("should handle result in useOne", () => {
    const source = `
      import { useOne } from "@refinedev/core";
      const { data } = useOne({
        id: 123
      });

      const post = data?.data ?? {};
    `;
    expect(transform(source)).toMatchSnapshot();
  });

  it("should handle renamed result in useOne", () => {
    const source = `
      import { useOne } from "@refinedev/core";
      const { data: postData } = useOne({
        id: 123
      });

      const post = postData?.data ?? {};
    `;
    expect(transform(source)).toMatchSnapshot();
  });

  it("should handle result with isLoading in useOne", () => {
    const source = /* tsx */ `
      import { useOne } from "@refinedev/core";

      export const EditProduct = () => {
        const { data, isLoading } = useOne({ id: 123 });

        if (isLoading) {
          return <div>Loading...</div>;
        }

        return (
          <div>
            <div>Product name: {data?.data.name}</div>
            <div>Product price: \${data?.data.price}</div>
          </div>
        );
      };
    `;

    expect(transform(source)).toMatchSnapshot();
  });
});
