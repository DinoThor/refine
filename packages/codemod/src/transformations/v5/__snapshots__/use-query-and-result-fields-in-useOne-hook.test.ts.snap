// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`use-query-and-result-fields-in-useOne-hook should handle renamed result in useOne 1`] = `
"
      import { useOne } from "@refinedev/core";
      const { result: postData } = useOne({
        id: 123
      });

      const post = postData ?? {};
    "
`;

exports[`use-query-and-result-fields-in-useOne-hook should handle result in useOne 1`] = `
"
      import { useOne } from "@refinedev/core";
      const { result } = useOne({
        id: 123
      });

      const post = result ?? {};
    "
`;

exports[`use-query-and-result-fields-in-useOne-hook should handle result with isLoading in useOne 1`] = `
"
      import { useOne } from "@refinedev/core";

      export const EditProduct = () => {
        const { result, query: {
          isLoading
        } } = useOne({ id: 123 });

        if (isLoading) {
          return <div>Loading...</div>;
        }

        return (
          <div>
            <div>Product name: {result?.name}</div>
            <div>Product price: \${result?.price}</div>
          </div>
        );
      };
    "
`;
